src/elements/stream_element.ts

```diff
@@ -75,7 +75,13 @@ export class StreamElement extends HTMLElement {
   }
 
   private get beforeRenderEvent() {
-    return new CustomEvent("turbo:before-stream-render", { bubbles: true, cancelable: true })
+    let properties: CustomEventInit = { bubbles: true, cancelable: true }
+
+    if (this.action !== 'remove') {
+      properties.detail = { newTemplate: this.templateContent }
```

Comment:
Something like `properties.detail = { streamElement: this }` would be more flexible.

Reply:
Great idea, thank you!